src/jdk.compiler/share/classes/com/sun/tools/javac/comp/ReflectMethods.java

```diff
@@ -192,7 +191,7 @@ public void visitMethodDef(JCMethodDecl tree) {
                 }
                 // create a static final field holding the op' string text.
                 // The name of the field is foo$op, where 'foo' is the name of the corresponding method.
-                classOps.add(opFieldDecl(methodName(tree), tree.getModifiers().flags, funcOp));
+                classOps.add(opFieldDecl(methodName(bodyScanner.symbolToErasedMethodRef(tree.sym)), tree.getModifiers().flags, funcOp));
```

Comment:
Optional comment (we can also leave it for another PR). I think the code would be a bit cleaner if the method name would be computed inside `opFieldDecl`, as that's what that method is about.